static/js/publisher/pages/RegisterSnap/RegisterSnap.tsx

```diff
@@ -0,0 +1,108 @@
+import { ReactNode, useState } from "react";
+import { Link } from "react-router-dom";
+import { Notification } from "@canonical/react-components";
+
+import RegisterSnapForm from "./RegisterSnapForm";
+
+import { filterAvailableStores } from "../../utils";
+
+import { useBrandStores } from "../../hooks";
+
+import type { RegistrationResponse } from "./local-types";
+
+function RegisterSnap(): ReactNode {
+  const [isSending, setIsSending] = useState<boolean>(false);
+  const [registrationResponse, setRegistrationResponse] =
+    useState<RegistrationResponse>();
+  const { data, isLoading } = useBrandStores();
+
+  const availableStores = filterAvailableStores(data || []);
+
+  return (
+    <div className="u-fixed-width">
+      {registrationResponse?.error_code === "already_registered" ||
+      registrationResponse?.error_code === "already_claimed" ? (
+        <>
+          <h1 className="p-heading--2">
+            <strong>{registrationResponse.snap_name}</strong> is already taken
+          </h1>
+
+          <Notification severity="caution">
+            Another publisher already registered{" "}
+            <strong>{registrationResponse.snap_name}</strong>. You can{" "}
+            <Link
+              to={`/register-name-dispute?snap-name=${registrationResponse.snap_name}&store=${registrationResponse.store}`}
+            >
+              file a dispute
+            </Link>{" "}
+            to request a transfer of ownership or register a new name below.
+          </Notification>
+        </>
+      ) : (
+        <h1 className="p-heading--2">Register snap</h1>
+      )}
+
+      {registrationResponse?.error_code === "name-review-required" && (
```

Comment:
I wonder if it would make sense to extract the error handling to separate component? Seems like the most of the JSX here is mapping error code to notifications.

Reply:
Makes sense. I'll extract it.